code/src/main/kotlin/com/expediagroup/sdk/v2/core/client/ApiClientApolloHttpEngine.kt

```diff
@@ -0,0 +1,64 @@
+package com.expediagroup.sdk.v2.core.client
+
+import com.apollographql.apollo.api.http.HttpRequest
+import com.apollographql.apollo.api.http.HttpResponse
+import com.apollographql.apollo.exception.ApolloNetworkException
+import com.apollographql.java.client.ApolloDisposable
+import com.apollographql.java.client.network.http.HttpCallback
+import com.apollographql.java.client.network.http.HttpEngine
+import com.expediagroup.sdk.v2.core.request.Request
+import com.expediagroup.sdk.v2.core.client.util.toApolloHeaders
+import okio.buffer
+import okio.source
+
+
+/**
+ * Implementation of the `HttpEngine` interface that uses an `ApiClient` instance to execute HTTP requests.
+ *
+ * This class provides a method to execute HTTP requests and handle the response or any exceptions that
+ * might occur during the request lifecycle. It leverages the `ApiClient` to configure and send the requests.
+ *
+ * @constructor Creates an `ApiClientApolloHttpEngine` with the specified `ApiClient`.
+ *
+ * @param client The instance of `ApiClient` used to execute the HTTP requests.
+ */
+class ApiClientApolloHttpEngine(
```

Comment:
```suggestion
class SdkApolloHttpEngine(
```

Reply:
We agreed to use `ApiClient` prefix for Apollo engine I believe, haven't we?

Plus, as per the naming convention, the `ApiClient` is a reference for the Google Api Client, not Apache or any concrete implementation.